pkg/testcase/secretsencrypt.go

```diff
@@ -44,40 +41,60 @@ func secretsEncryptOps(action, product, cpIP string, nodes []shared.Node) {
 	stdOutput, err := shared.SecretEncryptOps(action, cpIP, product)
 	Expect(err).NotTo(HaveOccurred(), "error: secret-encryption: "+action)
 	verifyActionStdOut(action, stdOutput)
-	if (action == "reencrypt") || (action == "rotate-keys") {
-		shared.LogLevel("DEBUG", "reencrypt op needs some time to complete - Sleep for 20 seconds before service restarts")
-		time.Sleep(20 * time.Second) // Wait for reencrypt action to complete before restarting services.
+
+	shared.LogLevel("debug", "secrets-encrypt ops need extra time to complete - Sleep for 30 seconds before service restarts")
+	time.Sleep(30 * time.Second)
+
+	// Restart Primary Etcd Node First
+	etcdIp, getError := shared.GetPrimaryEtcdIp(product)
```

Comment:
What if the primary is a control-plane node, will it work in that scenario?

Reply:
The primary node will most definitely be an etcd node. It can be only etcd or a combination of etcd,cp. That said, the GetPrimaryEtcdIp method reads the config.yaml file to get the primary node ip. The server IP we provide there - will always be the primary node ip - be it the config.yaml on a secondary etcd node/cp node/agent node. And a lack of the server line - means it is the primary node itself. So we can reference the config.yaml file from any node to still figure out the primary node.

Reply:
Yeah lets keep it simple `GetPrimaryNodeIP`